starlite/kwargs.py

```diff
@@ -267,30 +269,33 @@ def to_kwargs(self, connection: Union[WebSocket, Request]) -> Dict[str, Any]:
                 reserved_kwargs["socket"] = connection
             if "data" in self.expected_reserved_kwargs:
                 reserved_kwargs["data"] = self.get_request_data(request=cast(Request, connection))
+        connection_query_params = parse_query_params(
+            connection, {p.field_name for p in self.expected_query_params if p.is_array}
+        )
```

Comment:
Given that `to_kwargs()` is called in line with the request/response, I think it would be better to aggregate the set of array type query parameter names in `create_for_signature_model()` which is called during bootstrap. So instead of adding `is_array` to `ParameterDefinition` you could add `array_query_parameter_names` to `__slots__`, aggregate the set while iterating over the signature model fields in `create_for_signature_model()` and store it on the instance. Then, pass that already compiled set into `parse_query_params()` here.

Also, perhaps this should be evaluated before the `if self.expected_reserved_kwargs:` block, so that the mapping of query params injected if the `query` reserved kwarg is specified is also accurate for this case.

Reply:
Great point, thanks for the feedback. I completely missed the `expected_reserved_kwargs` - I added an additional test case to check the `query` kwarg is in line with expected parameters.